mango-core/src/main/java/org/calrissian/mango/types/TypeRegistry.java

```diff
@@ -117,7 +117,7 @@ public Object decode(String alias, U value) {
         if (encoder != null)
             return encoder.decode(value);
 
-        throw new TypeDecodingException("An unknown alias [" + value + "] was encountered");
+        throw new TypeDecodingException("An unknown alias [" + alias + "] was encountered while decoding value [" + value + "]. Valid aliases are " + String.join(", ", aliasMapping.keySet()));
```

Comment:
1. Trying toString sounds better than nothing. Maybe it'll confuse the issue by printing garbage, but maybe it'll be helpful for figuring out what data was being processed when this problem was encountered.
2. Mango uses the diamond operator all over the place, and that requires 1.7. I'll rewrite to use guava's Joiner instead.